pkg/config/nodetreemodel/config.go

```diff
@@ -706,11 +707,37 @@ func (c *ntmConfig) IsConfigured(key string) bool {
 	}
 	// if key is a leaf, we just check the source
 	if leaf, ok := curr.(LeafNode); ok {
-		return leaf.Source().IsGreaterThan(model.SourceDefault)
+		return leaf.Source().IsGreaterThan(model.SourceDefault) && leaf.Get() != nil
 	}
 
 	// if the key was an InnerNode we need to check all the inner leaf node to check if one was set by the user
-	return hasNoneDefaultsLeaf(curr.(InnerNode))
+	return hasNonDefaultLeaf(curr.(InnerNode))
+}
+
+func isInnerOrLeafWithNilValue(node Node) bool {
+	if node == missingLeaf {
+		return false
+	}
+	if _, isInner := node.(InnerNode); isInner {
+		return true
+	}
+	if leaf, isLeaf := node.(LeafNode); isLeaf {
+		return leaf.Get() == nil
+	}
+	return false
+}
+
+// HasSection returns whether the setting is
+func (c *ntmConfig) HasSection(key string) bool {
+	c.RLock()
+	defer c.RUnlock()
+
+	node := c.nodeAtPathFromNode(key, c.file)
+	if isInnerOrLeafWithNilValue(node) {
+		return true
+	}
+	node = c.nodeAtPathFromNode(key, c.envs)
+	return isInnerOrLeafWithNilValue(node)
```

Comment:
Why not check `root` ? We could have a config from FA or any other source.

Reply:
We don't want to return true if the node exists in the default layer, so we can't use root. But you're correct that the other layers should be checked, pushed a commit to fix it.